### ISSUE TITLE
Change "Locate audiobooks"

### ISSUE THREAD

**Issue by rmcrackan** [rmcrackan/Libation#1013]
This feature is confusing and it's usefulness is limited. Either

* move it to an 'advanced' section and include better instructions
* remove it

**Comment by CLHatch** [comment]
> This feature is confusing and it's usefulness is limited. Either
> 
> * move it to an 'advanced' section and include better instructions
> * remove it

It is a bit confusing... When I saw "Locate audiobook" I assumed it was to open File Explorer (in Windows, at least) to the location of the book files.  But it appears to be to SET the location of the book files, in case you've moved the folder?

**Comment by rmcrackan** [comment]
@CLHatch yeah, basically. Or if you've re-installed Libation. You can point to a folder and Libation will do a deep dive looking for folders and audiofiles with the audible id.

**Comment by CLHatch** [comment]
> @CLHatch yeah, basically. Or if you've re-installed Libation. You can point to a folder and Libation will do a deep dive looking for folders and audiofiles **with the audible id.**

The "audible id" part being the reason it's pretty much a requirement to have the audible id in the file or path, I'm guessing.  Which luckily I would have added anyways,

**Comment by rmcrackan** [comment]
Right. Libation needs a way to determine which file/s go to which audible book entry. Without using the id, it's a messy affair. Of course, if you don't specifically need Libation to know that you have that file, then no id needed and there's no problem.

Returning to the orig intent of this enhancement -- this is a very specific feature which has been wildly misunderstood by users over the years who reasonably assume it's something more general. Which is to say: I've made it too confusing.

**Comment by CLHatch** [comment]
> Right. Libation needs a way to determine which file/s go to which audible book entry. Without using the id, it's a messy affair. Of course, if you don't specifically need Libation to know that you have that file, then no id needed and there's no problem.
> 
> Returning to the orig intent of this enhancement -- this is a very specific feature which has been wildly misunderstood by users over the years who reasonably assume it's something more general. Which is to say: I've made it too confusing.

If you do keep the feature, I would suggest renaming it to more closely match what it does. Perhaps "Set location" or something like that?

**Comment by Mbucari** [comment]
@rmcrackan @CLHatch Actually, this feature specifically does _Not_ rely on filenames to determine the prodiuct ID. Since the beginning with AAXClean, all m4b files have a metadata field containing the ASIN (`CDEK`). Since as early as July 2023, all mp3s created by Libation also have a metadata field containing the ASIN (`AUDIBLE_ASIN`). Find Audiobooks opens and reads metadata from all mp3 and m4b files, and if it finds an ASIN it adds that book as "downloaded" within Libation and adds the files to the file path cache.

Maybe rename it to "Import Previously Liberated Autiobooks"?

**Comment by rmcrackan** [comment]
That's a much better name

**Comment by CLHatch** [comment]
> [@rmcrackan](https://github.com/rmcrackan) [@CLHatch](https://github.com/CLHatch) Actually, this feature specifically does _Not_ rely on filenames to determine the prodiuct ID. Since the beginning with AAXClean, all m4b files have a metadata field containing the ASIN (`CDEK`). Since as early as July 2023, all mp3s created by Libation also have a metadata field containing the ASIN (`AUDIBLE_ASIN`). Find Audiobooks opens and reads metadata from all mp3 and m4b files, and if it finds an ASIN it adds that book as "downloaded" within Libation and adds the files to the file path cache.
> 
> Maybe rename it to "Import Previously Liberated Autiobooks"?

Ah, I thought I had read somewhere in here that we "had" to have the ASIN somwhere in the rename string to let ABS find it.  Guess I was wrong on that, or that has changed. :)  Of course, I prefer to keep it in there somewhere anyways.  Both for me, and for "deobfuscation" (make sure we don't have folders with duplicate names between books).

**Comment by Mbucari** [comment]
@CLHatch where the ID in the file name comes into play is when you've downloaded an audiobook, and for some reason, that audiobook is not in Libation's file path cache. Libation will then search the Books directory for an audio file with that audible ID number in the file path. If it can't find it then it gives up. 

What this tool does is searches audio file metadata for audible ID numbers, and then it inserts all found audible audiobooks into Libation's file path cache so that they can be found later.
